src/infrastructure/healthchecks/HealthcheckRefreshJob.ts

```diff
@@ -0,0 +1,33 @@
+import type { CommonDependencies } from '../commonDiConfig.js'
+import { AbstractPeriodicJob } from '../jobs/AbstractPeriodicJob.js'
+import type { HealthcheckClass, SupportedHealthchecks } from './healthchecks.js'
+
+export class HealthcheckRefreshJob extends AbstractPeriodicJob {
+  public static JOB_NAME = 'HealthcheckRefreshJob'
+  private readonly healthCheckers: Record<SupportedHealthchecks, HealthcheckClass>
+
+  constructor(dependencies: CommonDependencies) {
+    super(
+      {
+        jobId: HealthcheckRefreshJob.JOB_NAME,
+        intervalInMs: 10000,
```

Comment:
🟠 @kibertoad should we also store when the last healthcheck ran? In case the refresh job starts failing for some reason, we'll keep serving stale results (false-positive), won't we?

Reply:
especially looking at `// if we don't yet have results, we consider that to be a success, in order to avoid arbitrary restarts` - that could also happen e.g. when they go stale in redis, expire and get deleted automatically (I'm not sure this is really how it works), in which case we'd just assume they're ok

Reply:
> should we also store when the last healthcheck ran

How would we use them?